src/transforms/concat.js

```diff
@@ -133,16 +132,28 @@ module.exports = packager => {
           );
         }
 
-        if (name.value === 'default' && mod.cacheData.isCommonJS) {
+        let node = replaceExportNode(mod.id, name.value, path);
+
+        // If the module has any CommonJS reference, it still can have export/import statements.
+        if (mod.cacheData.isCommonJS) {
           path.replaceWith(
-            DEFAULT_INTEROP_TEMPLATE({
-              MODULE: replaceExportNode(mod.id, name.value, path, false)
-            })
+            name.value === 'default'
+              ? DEFAULT_INTEROP_TEMPLATE({
+                  MODULE: t.isMemberExpression(node) ? node.object : node
+                })
+              : node
           );
+        } else if (t.isIdentifier(node)) {
+          path.replaceWith(node);
         } else {
-          path.replaceWith(replaceExportNode(mod.id, name.value, path));
+          let relativePath = relative(packager.options.rootDir, mod.name);
+
+          throw new Error(`${relativePath} does not export '${name.value}'`);
```

Comment:
What about a warning instead of an error, like how we do for `fs`? I have to comment this to get code using `readFileSync` working